packages/admin/cms-admin/src/pages/pageSearch/usePageSearch.ts

```diff
@@ -34,14 +34,77 @@ interface UsePageSearchOptions {
     onUpdateCurrentMatch: (pageId: string, pages: PageTreePage[]) => void;
 }
 
+function calculateMatchIndexOffset(fullPath: string[], pathNameStartsWithSlash: boolean): number {
+    fullPath.pop();
+
+    if (fullPath.length === 1 && !pathNameStartsWithSlash) {
+        return 1;
+    } else if (pathNameStartsWithSlash) {
+        return fullPath.join("/").length;
+    }
+    return fullPath.join("/").length + 1;
+}
+
+function searchPages(
+    query: string,
+    inorderPages: {
+        id: string;
+        parentId: string | null;
+        name: string;
+        ancestorIds: string[];
+        path: string;
+    }[],
+): PageSearchMatch[] {
+    const matches: PageSearchMatch[] = [];
+    const regex = new RegExp(`(${escapeRegExp(query)})`, "gi");
+    let match;
+
+    if (query.startsWith("/") && query.length > 1) {
```

Comment:
I don't think the search should differentiate between names and paths. If a page has the name /a/b/c and the path /a/b/c (very unlikely), both matches should be highlighted

Reply:
This also eliminates the need for `calculateMatchIndexOffset`

Reply:
The reason for this distinction is that the path does not necessarily have to match the name, and if someone starts the search string with "/", they are likely only looking for paths. It is possible to name a page "foo" and give it the path "/baa". Now, if a page called "foo-baa" exists with the path "/foo-baa", when someone searches for "baa", they probably do not want the "foo" page to be included in the results.